### PR TITLE
adding api tests for e2e testing

### PR DESCRIPTION
**Pull request by chajain**
Adding apis tests

### CODE REVIEW

**Comment by juadk** [comment]
Hello, JFYI, I've tested your new API tests inside a two nodes cluster built automatically by openQA (Testing Framework used at SUSE for operating system testing) and all looks good. 
I'll write a confluence page to explain the process asap.
image

**Comment by mbelur** [comment]
Looks good to me!
couple of things which can be addressed  as an enhancement later:
1. Could we use random names for the resources used in each test?
2. We should add cleanup of resources regardless of the test outcome whether success or fail.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,59 @@
...
+This is a harvester test framework which uses python pytest library. This framework heavily depends on the pytest fixtures which are defined in conftest.py
+
+## Pre-requisites
+-------------
+
+Before running these tests, bring up a virtual harvester environment following the instructions in repo https://github.com/harvester/ipxe-examples/tree/main/vagrant-pxe-harvester
+
+## Pytest Fixtures
+-------------
+They help us set up some helper code that should run before any tests are executed, and are perfect for setting-up resources that are needed by the tests. They are defined in coftest.py under the apis folder.
```
Comment (bk201) [inline README.md:12]:
```suggestion
They help us set up some helper code that should run before any tests are executed, and are perfect for setting-up resources that are needed by the tests. They are defined in conftest.py under the apis folder.
```
nit

**Comment by bk201** [comment]
LGTM!

Note some response codes and messages might be different after https://github.com/harvester/harvester/pull/919 is merged.
Since the document encourages users to use a virtualenv, It would be nice to have a tox file and add some lint check (like flake8) in the future.

**Review by bk201 — APPROVED** [review]

**Comment by guangyee** [comment]
> LGTM!
> 
> Note some response codes and messages might be different after [harvester/harvester#919](https://github.com/harvester/harvester/pull/919) is merged.
> Since the document encourages users to use a virtualenv, It would be nice to have a tox file and add some lint check (like flake8) in the future.

That's a good suggestion. Will add that in a follow-on PR.

**Review by guangyee — APPROVED** [review]
Need to add tox later. But this is a good starting point.

**Pull request merged**